deploy/charts/cert-manager/crds/challenges.yaml

```diff
@@ -78,6 +78,7 @@ spec:
           - dnsName
           - issuerRef
           - key
+          - solver
```

Comment:
The `solver` field was previously _not_ marked required - as part of this PR I've switched to actually make it a required field.

All code expects this field to be set (and returns an error if it is not), and in the one place where we set it (the 'orders' controller), we also do not allow it to be set to `nil`.

Any *external* consumers of this API may trip up on this change if they create a Challenge manually that does *not* specify the `spec.solver` field, however I don't think anyone is directly using the Challenge API anyway, so I think this is relatively safe for us to do.

The alternative would be to keep it marked as optional here and _convert_ only make it required in `v1beta1`, but I don't think this brings particularly much benefit in the absence of other consumers of this API that create Challenge resources without setting `spec.solver`.